website/docs/chapter-03/03_user_keyword.md

```diff
@@ -0,0 +1,434 @@
+
+:::
+
+:::K2[LO-3.3.5.3-2]
+
+Understand the role of embedded arguments in Behavior-Driven Development (BDD) style.
+
+:::
+
+::::
+
+
+In Robot Framework, **embedded arguments** allow the inclusion
+of arguments directly within the keyword name itself.
+This approach is particularly useful for creating
+**Behavior-Driven Development (BDD)**-style test cases or for
+making keyword names more readable and meaningful.
+
+With embedded arguments, placeholders are used within the keyword name,
+which are replaced by actual values when the keyword is executed.
+These arguments are written as scalar variables with dollar signs and curly braces,
+as shown in the following example:
+
+```robotframework
+*** Keywords ***
+The file '${file_name}' should contain '${expected_content}'
+    ${file_content} =    Get File    ${file_name}
+    Should Contain    ${file_content}    ${expected_content}
+```
+
+When this keyword is called, the placeholders `${file_name}`
+and `${expected_content}` are replaced by the actual values provided in the keyword call.
+For instance, in the following example,
+`${file_name}` is replaced with `server.log`
+and `${expected_content}` with `Successfully started`:
+
+```robotframework
+*** Test Cases ***
+Test File Content
+    Given the server log level is 'INFO'
+    When the server is started successfully
+    Then the file 'server.log' should contain 'Successfully started'
+```
+
+Quotes around the embedded arguments are treated as regular characters
+within the keyword name but can improve readability
+and help distinguish embedded arguments from the rest of the keyword name.
+
+Embedded arguments can become problematic when the keyword name becomes overly long or complicated.
+To address this, a mix of embedded arguments and regular arguments can be used.
+This approach can help manage more complex data structures and enhance readability.
+
+Example of mixed embedded and regular arguments:
+
+```robotframework
+*** Test Cases ***
+Embedded and normal arguments
+    Given the user is on the pet selection page
+    When the user adds    2     cat fish
+    And the user sets    3     dogs
+    And the user removes    1     dogs
+    Then the number of cat fish should be    2
+    And the number of dogs should be    count=2
+
+*** Keywords ***
+the number of ${animals} should be
+    [Arguments]    ${count}
+    ${current_count}    Get Animal Count    ${animals}
+    Should Be Equal As Numbers    ${current_count}    ${count}
+
+the user ${action}
+    [Arguments]    ${amount}   ${animal}
+    IF    '${action}' == 'adds'
+        Add Items To List    animal_list    ${animal}    ${amount}
+    ELSE IF    '${action}' == 'removes'
+        Remove Items From List    animal_list    ${animal}    ${amount}
+    ELSE IF    '${action}' == 'sets'
+        Set Amount To List    animal_list    ${animal}    ${amount}
+    ELSE
+        Skip    Test skipped due to invalid action
+    END
+```
```

Comment:
This test fails: 

`Keyword 'the user $(action)' expected 2 arguments, got 0.`